### PR TITLE
#485 Pick up locale requested by the client

### PR DESCRIPTION
**Pull request by Matasx**
TODO
- [ ] Documentation
- [x] Acceptance test
- [x] Testing

### CODE REVIEW

File: Modules/GenHTTP.Modules.I18n/Parsers/LanguageParser.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    /// <param name="acceptLanguageHeader"></param>
+    /// <returns></returns>
+    public static IEnumerable<string> ParseAcceptLanguageHeader(string? acceptLanguageHeader)
+    {
+        if (string.IsNullOrWhiteSpace(acceptLanguageHeader))
+        {
+            return [];
+        }
+
+        return acceptLanguageHeader
```
Comment (Matasx) [inline Modules/GenHTTP.Modules.I18n/Parsers/LanguageParser.cs:18]: On a second thought, I guess precompiled regex might be better choice...

**Review by Kaliumhexacyanoferrat — COMMENTED** [review]
Thanks for the PR, looks pretty good - just added some notes. 

In general, I have the feeling that the concern will do a lot of allocations. I cannot run dotMemory right now but especially the language parsing and the LINQ queries will allocate a lot of memory - typically the modules should try to do as few allocation as possible (for a simple "hello world" example the whole server will only allocate objects for the request, the response objects and some strings). This might not matter for single requests but if you process hundreds or even thousands per second this will be significant (I guess around 5% slow down with this concern in place). This is also the reason why some of the code of GenHTTP probably looks not as elegant as it could.

Don't know whether you have an IDEA Ultimate license - there is dotMemory in the suite which is super helpful to analyze the allocations. I have a free Open Source license from JetBrains for this project, but I am not sure whether I can extend this for another contributor currently as they are running for a whole year (until May 2025).

File: Modules/GenHTTP.Modules.I18n/Types.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using GenHTTP.Api.Protocol;
+using System.Globalization;
+
+namespace GenHTTP.Modules.I18n;
+
+/// <summary>
+/// Delegate to select the cultures for a given request.
```
[the remark this answers is not in the document]
Comment (Kaliumhexacyanoferrat) [reply Modules/GenHTTP.Modules.I18n/Types.cs:7]: So "Delegate to extract the accepted cultures for a given request"

**Comment by Matasx** [comment]
Thank you. The easy parts are updated. I will now focus on the memory consumption and async support. I don't have jetbrains licence, but the memory profiler bundled with MSVS Pro is also somewhat useful. I'll see what I can do. 😸

**Pull request merged**